remote/remote.go

```diff
@@ -356,7 +356,7 @@ func (i *Image) CheckReadWriteAccess() bool {
 	if err != nil {
 		return false
 	}
-	return remote.CheckPushPermission(ref, i.keychain, http.DefaultTransport) == nil
+	return i.CheckReadAccess() && remote.CheckPushPermission(ref, i.keychain, http.DefaultTransport) == nil
```

Comment:
I agree that "write only" access would be silly. But this doesn't do any harm...

Reply:
Actually, because we changed the method signature to `CheckReadWriteAccess()`, actually it should also validate the Read access, right?

Reply:
I have some thoughts about the `CheckReadWriteAccess` and `CheckReadAccess` functions:
* Maybe we should move `remote.CheckPushPermission(ref, i.keychain, http.DefaultTransport) == nil` to a private function `checkWriteAccess` so the code will be clearer.
* Should `CheckReadWriteAccess` return two booleans or a struct with two booleans so it'll be clearer what kind of access (if any) the image has?
* Although both functions are public, I think it's redundant to run almost the same tests as part of `remote_test.go`. Should we only keep those under `CheckReadWriteAccess`? And if not, ,maybe we can create a function to test all of these scenarios together (the function will get the function that should run on the image - either `CHeckReadWriteAccess` or `CheckReadAccess`).

Reply:
Regarding your third comment, I thought something similar [here](https://github.com/buildpacks/imgutil/pull/125#discussion_r671434624). This is a preview of how it will look if I move the `CheckReadAccess` assertions to the `CheckReadWriteAccess` scenarios

```
when("image has read and write access", func() {
    it("returns true", func() {
	    image, err := remote.NewImage(customRegistry.RepoName(readWriteImage), authn.DefaultKeychain)
	    h.AssertNil(t, err)
	    h.AssertEq(t, image.CheckReadWriteAccess(), true)
	    h.AssertEq(t, image.CheckReadAccess(), true)
    })
})

when("image has read access but no write access", func() {
    it("returns true for CheckReadAccess() and false for CheckReadWriteAccess()", func() {
	    image, err := remote.NewImage(customRegistry.RepoName(onlyReadImage), authn.DefaultKeychain)
	    h.AssertNil(t, err)
	    h.AssertEq(t, image.CheckReadWriteAccess(), false)
	    h.AssertEq(t, image.CheckReadAccess(), true)
    })
})

when("image doesn't have read access but has write access", func() {
    it("returns false", func() {
	    image, err := remote.NewImage(customRegistry.RepoName(onlyWriteImage), authn.DefaultKeychain)
	    h.AssertNil(t, err)
	    h.AssertEq(t, image.CheckReadWriteAccess(), false)
	    h.AssertEq(t, image.CheckReadAccess(), false)
    })
})

when("image doesn't have read nor write access", func() {
    it("returns false", func() {
	    image, err := remote.NewImage(customRegistry.RepoName(noAccessImage), authn.DefaultKeychain)
	    h.AssertNil(t, err)
	    h.AssertEq(t, image.CheckReadWriteAccess(), false)
	    h.AssertEq(t, image.CheckReadAccess(), false)
    })
})
```

Reply:
I'm wondering if it doesn't make the code confusing, because all of these `when`s will be under `when("CheckReadWriteAccess"`, right?
I thought about completely deleting the `CheckReadAccess` calls because they're part of the `CheckReadWriteAccess` code, but then I think that we need to change a bit the `CheckReadWriteAccess` interface to return what kind of access we have.